webpack/assets/javascripts/react_app/components/AutoComplete/index.js

```diff
@@ -23,18 +23,42 @@ class AutoComplete extends React.Component {
       'handleInputFocus',
       'getResults',
       'windowKeyPressHandler',
-      'unableHTMLAutocomplete',
+      'disableHTMLAutocomplete',
       'handleKeyDown',
     ]);
     this._typeahead = React.createRef();
     debounceMethods(this, 500, ['handleLoading']);
+    // Used to pass _typeahead into the static function getDerivedStateFromProps.
+    this.state = {
```

Comment:
I have mixed feelings about this, It seems like a big anti pattern to store a ref inside a state, plus having a two source of truth here might cause bugs.

Reply:
because of the static function, I had to find a way to pass the ref
so it isn't really another source of truth, it is just part of the mechanism of `getDerivedStateFromProps` which in my opinion is a lot more work than `componentWillReceiveProps` that wasn't static... though we can't use it anymore

Reply:
So typehead is stored as an internal state and as a ref on this. Which means you have two pointers on the same object, which defenatly might cause bugs. I don’t think the state meant for that, looks like an anti pattern to me.

Reply:
`component did mount` worked perfectly. No need for this state engagement anymore